bittensor/utils/networking.py

```diff
@@ -0,0 +1,125 @@
+import miniupnpc
+import os
+import requests
+import urllib
+
+from loguru import logger
+
+class ExternalIPNotFound(Exception):
+    """ Raised if we cannot attain your external ip from CURL/URLLIB/IPIFY/AWS """
```

Comment:
we are starting to scatter exceptions everywhere, can we put them all into exceptions.py or something?